Sage/carbon
# fix(filterable-select): ensure onchange called when deleting text

In a controlled FilterableSelect, onChange should be called after all interactions with the text input, including deleting a character when this results in a match. This does not happen with the current implementation.

### Proposed behaviour

Pressing the delete key should call `onChange` with the appropriate value when it results in the textbox value matching one of the options.

### Current behaviour



Pressing delete never triggers `onChange` (or calls it with an empty string).

### Checklist




#### QA


### Additional context


This issue was found while investigating #6493 , although it doesn't actually fix the issue demonstrated there, and as explained in my comment when closing the issue, I think we shouldn't bake the requested functionality into the component.

### Testing instructions



The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/select-carbon-forked-5vksn6?file=/src/CustomSelect.js

To reproduce:

- select “item0”
- focus the textbox and type “1" (or any character) at the end
- delete the just-added character

The bug is that the `value` which is updated in the `onChange` is still empty, as seen in the text above the input - even though other sequences which end in the textbox content being `item0` result in the value being `0`.
